### PR TITLE
Create a new Island for WeatherWidget and hook it up to our Weather API

### PR DESCRIPTION
**Pull request by jamesgorrie**
Part of #6378 

## What does this change?

Adds `Weather.importable` island and hooks it up to fronts.

We've had to make some change to the weather API types, and I've tried to keep them aligned in the components to avoid cognitive load. There might be some niceness of making the API sue nicer names, but maybe v2.

## Why?
We ❤️ the weather.

## Screenshots

| Before      | After      |
| ----------- | ---------- |
| ![before][] | ![after][] |

[before]: https://github.com/guardian/dotcom-rendering/assets/21217225/24ef0e3b-d569-4e1a-af50-291202c5e8e7
[after]: https://github.com/guardian/dotcom-rendering/assets/21217225/e466690c-f32f-4bf0-baee-8aa0c9901710

### CODE REVIEW

File: dotcom-rendering/src/components/Weather.importable.tsx [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+		country: string;
+	};
+	weather: WeatherData;
+	forecast: [WeatherData, WeatherData, WeatherData, WeatherData];
+};
+
+type Props = { edition: FEFrontConfigType['edition'] };
+
+export const WeatherImportable = ({ edition }: Props) => {
+	const { /*loading,*/ data /*error*/ } = useApi<WeatherApiData>(
```
Comment (jamesgorrie) [inline dotcom-rendering/src/components/Weather.importable.tsx:29]: Just put this here for future use... we can remove if we don't want it.

File: dotcom-rendering/src/components/WeatherWidget.tsx
Comment (AshCorr) [inline dotcom-rendering/src/components/WeatherWidget.tsx:1]: Yes I've renamed it back to WeatherWidget again 😅 

Didn't want to name the Island something other than `Weather` which would have conflicted a bit with this file.

**Comment by jamesgorrie** [comment]
> Yes I've renamed it back to WeatherWidget again

Not that I mind for this PR - but I think a naming convention we've used previously is:
`<WeatherData><Weather {...data} /><WeatherData>`.

I suppose why I liked `Weather` being the current one is that's what it is - and Widget feels really Y2K.

File: dotcom-rendering/src/lib/useApi.tsx [hunk 2]
```diff
@@ -16,7 +16,7 @@ const fetcher = (init?: RequestInit) => (url: string) =>
 		.then(checkForErrors)
 		.then((res) => res.json());
 
-interface ApiResponse<T> {
+export interface ApiResponse<T> {
```
Comment (jamesgorrie) [inline dotcom-rendering/src/lib/useApi.tsx:19]: You could leave this un-exported, and have the method infer the type.

File: dotcom-rendering/src/lib/useWeather.ts
[the remark this answers is not in the document]
Comment (jamesgorrie) [reply dotcom-rendering/src/lib/useWeather.ts:1]: It also creates an abstraction which could be inferred as that it is reusable. Maybe it is, but I'd probably say it's very specific to the component using it.

**Review by jamesgorrie — COMMENTED** [review]
This looks great. A few things I've added around co-location of code and naming, but that's all changeable easily if we want to in future PRs.

File: dotcom-rendering/src/lib/useApi.tsx (hunk 2 above)
Comment (AshCorr) [reply dotcom-rendering/src/lib/useApi.tsx:19]: I think our linting rules require the return type on an exported function to be explicit but I'll give it a go

File: dotcom-rendering/src/lib/useWeather.ts
[the remark this answers is not in the document]
Comment (AshCorr) [reply dotcom-rendering/src/lib/useWeather.ts:1]: Don't know why I didn't just export them from Weather... doh.

File: dotcom-rendering/src/components/WeatherWidget.tsx [hunk 3]
```diff
@@ -171,50 +172,43 @@ const ExternalLinkIcon = () => (
...
 }
 
-export const Weather = ({ location, now, forecast, edition }: WeatherProps) => {
+const getForecastForHourOffset = (
+	forecast: WeatherForecast,
+	offsetHours: 3 | 6 | 9 | 12,
+) => {
+	const dateTime = new Date(Date.now());
+	dateTime.setHours(dateTime.getHours() + offsetHours);
+	dateTime.setMinutes(0);
```
Comment (arelra) [inline dotcom-rendering/src/components/WeatherWidget.tsx:192]: Just double checking if this should be timezone aware?

Comment (AshCorr) [reply dotcom-rendering/src/components/WeatherWidget.tsx:192]: Ah crumbs, you're right.

I think Frontend is missing a field in its response to tell us what time these forecasts should be, I tried to workaround that but with timezones i don't think thats possible.

**Comment by AshCorr** [comment]
For some reason the Chromatic snapshots of the Weather widget are blank, but it seems like this was the case before this PR so I'm going to leave it for now.

https://www.chromatic.com/review?appId=63e251470cfbe61776b0ef19&number=8096&type=linked&view=changes

File: dotcom-rendering/src/layouts/FrontLayout.tsx [hunk 4]
```diff
@@ -110,6 +111,41 @@ const decideAdSlot = (
...
+		['uk/commentisfree', 'au/commentisfree'].includes(front.config.pageId)
+	) {
+		return <CPScottHeader />;
+	}
+
+	// show weather?
+	if (
+		front.config.switches['weather'] &&
+		['uk', 'us', 'au', 'international', 'europe'].includes(
+			front.config.pageId,
```
Comment (mxdvl) [inline dotcom-rendering/src/layouts/FrontLayout.tsx:130]: We’ve got a pattern for this if you’re keen! 😀 

```ts
import {guard} from './lib/guard';

const isEditionFrontPage = guard(['uk', 'us', 'au', 'international', 'europe'] as const);
```

**Pull request merged**